### PR TITLE
AI-VAD bbox post-processing

### PR DESCRIPTION
**Pull request by djdameln**
# Description

- This PR adds bounding box post-processing operations to the AI-VAD implementation. These operations were not mentioned in the paper, but were included in the original implementation.
- The post-processing operations consist of:
  - filtering by class label (some datasets benefit from only including person-regions)
  - filtering by bbox surface area
  - Removing overlapping bounding boxes
  - Adding regions that are undetected by the mask rcnn region extractor, but deemed to belong to the foreground by comparing pixel movement between consecutive video frames.
- Several configurable parameters were added to control the behaviour of the bbox post-processing
- All combined, these changes increase the image AUROC on ped2 dataset from 0.917 to 0.960

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

File: src/anomalib/models/ai_vad/regions.py [hunk 1]
```diff
@@ -17,21 +23,213 @@ class RegionExtractor(nn.Module):
...
+        pixel_diff = torch.abs(first_frame - last_frame)
+        pixel_diff = rgb_to_grayscale(pixel_diff).squeeze(1)
+
+        # apply binary threshold to the diff
+        foreground_map = (pixel_diff > binary_threshold / 255).int()
+
+        # remove regions already detected by region extractor
+        boxes_list = [im_regions["boxes"] for im_regions in regions]
+        boxes_list = [
+            clip_boxes_to_image(boxes + Tensor([-2, -2, 2, 2]).to(boxes.device), foreground_map.shape[-2:])
```
Comment (samet-akcay) [inline src/anomalib/models/ai_vad/regions.py:109]: Would it be possible to add a comment here to explain why we have `Tensor([-2, -2, 2, 2])`

**Review by samet-akcay — COMMENTED** [review]
Thanks, only a single comment.

**Review by samet-akcay — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
+- Add bounding box post-processing operations in AI-VAD implementation (<https://github.com/openvinotoolkit/anomalib/pull/1103>)
```
Comment (ashwinvaidya17) [inline CHANGELOG.md:21]: I think this will cause conflict with the latest changes in main. We should probably put this under unreleased

File: src/anomalib/models/ai_vad/lightning_model.py [hunk 3]
```diff
@@ -37,6 +37,12 @@ class AiVad(AnomalyModule):
     def __init__(
         self,
         box_score_thresh: float = 0.8,
+        persons_only: bool = False,
```
Comment (ashwinvaidya17) [inline src/anomalib/models/ai_vad/lightning_model.py:40]: I think the docstring needs to be updated

File: src/anomalib/models/ai_vad/regions.py [hunk 4]
```diff
@@ -17,21 +23,213 @@ class RegionExtractor(nn.Module):
...
         self.backbone = maskrcnn_resnet50_fpn_v2(weights=weights, box_score_thresh=box_score_thresh, rpn_nms_thresh=0.3)
 
-    def forward(self, batch: Tensor) -> list[dict]:
+    def forward(self, first_frame: Tensor, last_frame: Tensor) -> list[dict]:
         """Forward pass through region extractor.
 
         Args:
-            batch (Tensor): Batch of input images of shape (N, C, H, W)
+            first_frame (Tensor): Batch of input images of shape (N, C, H, W) forming the first frames in the clip
+            batch (Tensor): Batch of input images of shape (N, C, H, W) forming the last frame in the clip
```
Comment (ashwinvaidya17) [inline src/anomalib/models/ai_vad/regions.py:53]:
```suggestion
            last_frame (Tensor): Batch of input images of shape (N, C, H, W) forming the last frame in the clip
```

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks! I have a few minor comments

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks
